test/langtools/jdk/internal/shellsupport/doc/SinceCheckerHelper.java

```diff
@@ -0,0 +1,317 @@
+import com.sun.source.util.JavacTask;
+import com.sun.tools.javac.code.Flags;
+import com.sun.tools.javac.code.Symbol;
+import jdk.internal.shellsupport.doc.JavadocHelper;
+
+import javax.lang.model.element.*;
+import javax.lang.model.util.ElementFilter;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+import javax.tools.*;
+import java.io.IOException;
+import java.lang.Runtime.Version;
+import java.net.URI;
+import java.nio.file.*;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class SinceCheckerHelper {
+
+    //these are methods that were preview in JDK 13 and JDK 14, before the introduction
+    //of the @PreviewFeature
+    static final Set<String> LEGACY_PREVIEW_METHODS = Set.of(
```

Comment:
IIRC we will need a bit more of this when we include `java.compiler` and `jdk.compiler`. I think I would create a separated section for detection of "isPreview", which would include this set (which may need to be changed to a map, saying in which versions the given element was a preview).